activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -509,7 +512,8 @@ def index_name_length
         end
 
         # Maps logical Rails types to PostgreSQL-specific data types.
-        def type_to_sql(type, limit = nil, precision = nil, scale = nil)
+        def type_to_sql(type, limit = nil, precision = nil, scale = nil, array = nil)
```

Comment:
it works too, although i think rails style guide tend to discourage `tap` usage, especially in an already complex method